guard/src/migrate/parser.rs

```diff
@@ -98,37 +95,49 @@ impl Display for PropertyComparison {
     }
 }
 
+#[derive(Ord, Eq, PartialEq, PartialOrd, Debug, Clone, Hash)]
+pub(crate) struct TypeName {
+    pub type_name: String
+}
+impl Display for TypeName {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.type_name.to_lowercase().replace("::", "_"))
+    }
+}
+
 #[derive(PartialEq, Debug, Clone, Hash)]
 pub(crate) struct BaseRule {
-    pub(crate) type_name: String,
+    pub(crate) type_name: TypeName,
     pub(crate) property_comparison: PropertyComparison,
     pub(crate) custom_message: Option<String>
 }
-
 impl Display for BaseRule {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match &self.custom_message {
             Some(message) => {
-                write!(f, "%{}.{} <<{}>>", self.type_name.to_lowercase().replace("::", "_"), self.property_comparison, message)
+                write!(f, "{} <<{}>>", self.property_comparison, message)
             },
             None => {
-                write!(f, "%{}.{}", self.type_name.to_lowercase().replace("::", "_"), self.property_comparison)
+                write!(f, "{}", self.property_comparison)
             }
         }
     }
 }
 
 #[derive(PartialEq, Debug, Clone, Hash)]
 pub(crate) struct ConditionalRule {
-    pub type_name: String,
+    pub type_name: TypeName,
     pub when_condition: PropertyComparison,
     pub check_condition: PropertyComparison
 }
 impl Display for ConditionalRule {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        writeln!(f, "when %{}[ {} ] not EMPTY {{", self.type_name.to_lowercase().replace("::", "_"), self.when_condition);
-        writeln!(f, "\t\t%{}[ {} ].{}", self.type_name.to_lowercase().replace("::", "_"), self.when_condition, self.check_condition);
-        write!(f, "\t}}")
+        writeln!(f, "when %{}[ {} ] not EMPTY {{", self.type_name, self.when_condition);
```

Comment:
If i understand the change right, you already have an external `%aws_ec2_volume { ... }` block. Inside the block `this` already refers each `AWS::EC2::Volume`, does this `when` check need the filter part? It seems like we will generate

```
      %aws_ec2_volume {
            %aws_ec2_volume[ Properties.Size > 500 ] NOT EMPTY {
                  %aws_ec2_volume[ Size > 500 ].Encrypted == false
            }
      }
```

We should just generate 
```
      %aws_ec2_volume {
             when Properties.Size > 500 {
                     Properties.Encrypted == false
             }
      }
```
Easier to comprehend as well.